app/models/playlist.rb

```diff
@@ -99,6 +99,14 @@ def valid_token?(token)
     access_token == token && visibility == Playlist::PRIVATE_WITH_TOKEN
   end
 
+  def clips_with_section_proxies
+    cached_clips = clips.to_a
+    section_ids = cached_clips.collect(&:master_file_id).uniq
+    sections = SpeedyAF::Proxy::MasterFile.where("id:#{section_ids.join(' id:')}")
+    cached_clips.map {|c| c.master_file = sections.find { |mf| mf.id == c.master_file_id }}
```

Comment:
Should this be `map!` instead? Or does stuff get persisted anyway even with not doing the change in place?